### DISCUSSION TITLE
Allow Vercel Integrations to inject <script> tag at build time

### DISCUSSION THREAD

**Discussion by brunocrosier** [vercel/vercel#4479] [Ideas] [answered]
It would be pretty cool to allow Vercel Integrations to add a <script> tag to the header for example 

This would allow for one-click setup of pretty much every third-party service (Analytics, Live Chat etc.)

Something similar to [CloudFlare Apps](https://www.cloudflare.com/apps/)

Would this be possible ? I think it would open up a lot of doors for integrations

**Comment by styfle** [comment, marked as answer]
Thanks for the feedback, I'll pass it along to the team!

Today, you can achieve a similar result with an [integration](https://vercel.com/integrations) and the [environment variable ui](https://vercel.com/blog/environment-variables-ui). Then you can add a `<script>` tag wherever you need in your code and reference the environment variable there.

**Reply by brunocrosier** [reply]
Thanks @styfle . Yeah for sure we can already add `<script>` tag as you suggested but it would be cool for Vercel Integrations to allow for a one-click install of third party services, without having to touch code. 

So the process would be something like:

1. Ship code
2. Go to vercel.com/integrations or vercel.com/apps etc.
3. Find third-party integration (eg. Google Analytics)
4. Click "enable" and type in unique Google Analytics ID
5. Project rebuilds automatically, with added HTML from the integration / app. The same applies to future builds (while integration is enabled)

Google Analytics is now integrated without having to add any code, and can be easily removed right from within the Vercel dashboard. 

I feel that enabling this kind of plug and play would be hugely beneficial to growing the Vercel ecosystem

**Reply by styfle** [reply]
Makes sense!

We'll probably start with Next.js plugins for these types of features so you could run locally with `next dev` and make sure it works properly before deploying.

**Reply by brunocrosier** [reply]
@3lonious i ... don't think you're understanding my initial point ? you can of course already add script tags in your application. If you're using Next.js then you need to add it like this:

```javascript
import Head from "next/head"

export default () => (
<>
<Head>
<script async src="http://example.com/script.js" />
<script
                dangerouslySetInnerHTML={{
                  __html: `
                  // insert javascript here
                  `,
                }}
              />
</Head>

<h1>my component</h1>

</>
)

```

Also please tone your anger down, it's not a good look.

**Reply by PTBYSR** [reply]
Is it possible to edit code straight from the vercel dashboard without needing to go to the code base? I want to add a scripts for GTM but i don't have access to the code base just the vercel dashboard

**Reply by amyegan** [reply]
Hi @PTBYSR. No, there is no access to edit code straight from your Vercel Dashboard. Code security is handled by the code repository, so you would need to get access to the repository to make changes.

**Comment by 3lonious** [comment]
ok my bad ya it took me awhile to figure it out, but i got it
